docs/ref/models/fields.txt

```diff
@@ -1116,7 +1116,7 @@ Django also defines a set of fields that represent relations.
 ``ForeignKey``
 --------------
 
-.. class:: ForeignKey(othermodel, [**options])
+.. class:: ForeignKey(othermodel, on_delete, [**options])
```

Comment:
That's a really good point. The current second argument is `to_field`, which (judging by the number of unfixed bugs with it) is not frequently used, and when it is used I'm guessing it's usually used by keyword, for clarity. But technically changing the order of arguments is still a backwards-incompatible change that ought to have a deprecation path.

I think we can do it now, and have a deprecation path. All arguments to `on_delete` should be callable, and no argument to `to_field` can be callable. So it should be possible to switch the order, but automatically detect if we get a value for `on_delete` that was intended for `to_field`, and in that case issue a warning and adjust accordingly.